### ISSUE TITLE
Identities API doesn't work

### ISSUE THREAD

**Issue by alexkokkinos** [Microsoft/vsts-node-api#131]
The Identities API has no corresponding REST API documented, and when an attempt is made to make a basic call (such as `listGroups()`), an error is returned:

```
Error:Failed to find api location for area: IMS id:5966283b-4196-4d57-9211-1b68f41ec1c2
```

https://stackoverflow.com/questions/47682163/vsts-node-api-cant-use-the-identities-api

https://developercommunity.visualstudio.com/content/problem/163711/identities-api-in-vsts-node-api-preview-version.html

**Comment by jakekrammer-axosoft** [comment]
Same issue.

**Comment by bryanmacfarlane** [comment]
We have someone investigating.

@stephenmichaelf - this is likely due to the thread I added you around big S service support that @keithrob started.  It fails calling other services outside of TFS services (build, wit, vc) such as RM and identity which are other big S services unless you know the DNS FQDN those services resolve to (customers shouldn't need to know that - just the account url).

**Comment by jakekrammer-axosoft** [comment]
I'm trying to find a way to get the authenticated user's information from VSTS using this package. After a user has authenticated in my app using their personal access token, what API call should I be using to get their display name and avatar?

My original guess was either the IdentitiesAPI's `getSelf()` or ProfileAPI's `getUserDefaults(includeAvatar=true)`, but both of those API's appear to be suffering from issue https://github.com/Microsoft/vsts-node-api/issues/131. 

Code snippet:
```javascript
import vsts from 'vso-node-api';

const browserHost = 'https://krammerinc.visualstudio.com';
const authHandler = vsts.getPersonalAccessTokenHandler(oauthToken);
const connection = new vsts.WebApi(this.browserHost, authHandler);
const vstsIdentitiesApi = connection.getIdentitiesApi();

vstsIdentitiesApi.getSelf()
  .then((self) => {
    console.log(self);
  });
```

An example of the errors I'm encountering: 
![screen shot 2018-01-30 at 9 46 34 am](https://user-images.githubusercontent.com/31415703/35586537-4c4cbd48-05b8-11e8-919b-1f0e10438cc8.png)

**Comment by bryanmacfarlane** [comment]
@jakekrammer-axosoft - Thanks!

**Comment by stephenmichaelf** [comment]
@alexkokkinos This will be resolved by #139 

@jakekrammer-axosoft If you check in samples.ts there is an example of getting the authenticated user's information [here](https://github.com/Microsoft/vsts-node-api/blob/d467665e8ce692b5d9f10409a928a9e406c2428a/samples/common.ts#L49). Can you let me know if that unblocks you? Thanks!

**Comment by stephenmichaelf** [comment]
@alexkokkinos Could you get the latest 6.3.0-preview version and see if this works now?

**Comment by bryanmacfarlane** [comment]
to get that you need to run

```
npm install vso-task-lib@preview
```

**Comment by stephenmichaelf** [comment]
vso-node-api* ?

**Comment by alexkokkinos** [comment]
@stephenmichaelf I'll update my project and try it out, thanks!

**Comment by alexkokkinos** [comment]
@stephenmichaelf @bryanmacfarlane The 6.3.0-preview version doesn't work at all for me, only parts of our project utilizing REST API calls instead of the VSTS Node API work (like getting/creating Area paths).

The project we've built out (an Express API used to interface VSTS with various internal systems) utilizes the following:

CoreApi
WorkApi
GitApi
WorkItemTrackingApi

All of the above works with 6.2.8-preview, but it appears that all of those of pieces stop working when I configure my package.json with 6.3.0-preview, `npm install`, and run again.

I can consistently switch between these two versions to break and fix the project.

All calls I have tested so far result in errors similar to this:
`2018-02-13T21:32:15.400Z - error: CoreApi.getTeams is not a function`

So, I got the same error out of an attempt to use the Identities API (I've tried `getIdentities` and `getSelf`):
```
Promise {
  <rejected> TypeError: IdentitiesApi.listGroups is not a function
    at Object.readIdentities (C:\Users\alex\vsts\DevOps.Vsts.API\lib\vsoFunctions.js:31:37)
    at TeamsController.getSelf (C:\Users\alex\vsts\DevOps.Vsts.API\api\v1\teams.js:18:22)
    at TeamsController._router.get (C:\Users\alex\vsts\DevOps.Vsts.API\api\v1\teams.js:9:52)
    at Layer.handle [as handle_request] (C:\Users\alex\vsts\DevOps.Vsts.API\node_modules\express\lib\router\layer.js:95:5)
    at next (C:\Users\alex\vsts\DevOps.Vsts.API\node_modules\express\lib\router\route.js:137:13)
    at Route.dispatch (C:\Users\alex\vsts\DevOps.Vsts.API\node_modules\express\lib\router\route.js:112:3)
    at Layer.handle [as handle_request] (C:\Users\alex\vsts\DevOps.Vsts.API\node_modules\express\lib\router\layer.js:95:5)
    at C:\Users\alex\vsts\DevOps.Vsts.API\node_modules\express\lib\router\index.js:281:22
    at Function.process_params (C:\Users\alex\vsts\DevOps.Vsts.API\node_modules\express\lib\router\index.js:335:12)
    at next (C:\Users\alex\vsts\DevOps.Vsts.API\node_modules\express\lib\router\index.js:275:10) }
```

We aren't using TypeScript, it's pure Node/JS/Express. VSCode seems to have the matching relevant information about these functions that supposedly don't exist:

![image](https://user-images.githubusercontent.com/22206826/36175383-3598f7bc-10dd-11e8-9668-d52742fd4f16.png)

**Comment by stevek-axo** [comment]
@alexkokkinos It looks like as of the new preview, the return values of the `WebApi` API getters are Promises (e.g. `connection.getIdentitiesApi()`) The functions don't exist because you're trying to call it on the Promise object that gets returned.

@stephenmichaelf @bryanmacfarlane As of  I'm still noticing the same error message popping up that @jakekrammer-axosoft previously posted in this thread (`Failed to find api location for area: IMS id: 4bb02b5b-c120-4be2-b68e-21f7c50a4b82`) on all the calls I've tried to IdentitiesApi.

**Comment by bryanmacfarlane** [comment]
@stevek-axo - Correct.  We bumped the minor version because as of 6.2.8, 6.x was still under preview and we were waiting for the Big S service support (RM, etc...) to switch to release.  In order to resolved the url via location services we had to turn those functions into async.

See the samples not the await: https://github.com/Microsoft/vsts-node-api/blob/master/samples/build.ts#L11

**Comment by stephenmichaelf** [comment]
@stevek-axo Can you give some sample code of how you're using the IdentitiesApi, and which methods you are calling? Were these methods ever working? 

I think the IdentitiesApi needs a deployment level URL instead of a collection level URL(this is our fault, it's not documented well). As an experiment could you try doing <account>.visualstudio.com (without the collection name at the end of the URL) and see if it works?

**Comment by alexkokkinos** [comment]
@stevek-axo Ah that does make sense with the change to async, I'll have to make some changes to our project and try it out again.

**Comment by stevek-axo** [comment]
@stephenmichaelf I'm trying to stand up some integrations with VSTS on a project I am working on with @jakekrammer-axosoft. I was hoping to use the IdentitiesApi to fetch information for the Git tree such as usernames and profile pictures. I've been trying to use `identitiesApi.getSelf()` with personal access tokens for authentication. It hasn't worked ever since I've started working with `vsts-node-api` last month.

I've been trying on a `.visualstudio.com` URL, to no avail.

Here's a sample Node script demonstrating the problem. Token has full access to all scopes.
```javascript
const vsts = require('vso-node-api');

const accessToken = 'REDACTED';
const url = 'https://REDACTED.visualstudio.com';

const accessTokenHandler = vsts.getPersonalAccessTokenHandler(accessToken);
const connection = new vsts.WebApi(url, accessTokenHandler)

connection.getIdentitiesApi()
  .then((identitiesApi) => {
    identitiesApi.
    return identitiesApi.getSelf();
  })
  .then((self) => {
    console.log(self);
  })
  .catch((error) => {
    console.log(error);
  });
```

**Comment by stephenmichaelf** [comment]
@stevek-axo There is an interface called [IdentityRef](https://github.com/Microsoft/vsts-node-api/blob/bdef7c9aeec1bbc8e80d42b0ce48b8269795f7b2/api/interfaces/common/VSSInterfaces.ts#L132) that has displayName and imageUrl. I think those are what you are looking for.

In terms of how to get an IdentityRef, if you are looking at Git specifically, you can check the [GitInterfaces](https://github.com/Microsoft/vsts-node-api/blob/bdef7c9aeec1bbc8e80d42b0ce48b8269795f7b2/api/interfaces/GitInterfaces.ts). Depending on what information you are pulling back you should have direct access to any identity information through the Git response.

**Comment by stephenmichaelf** [comment]
Also if you use the git api:

```
const gitApi = await vsts.getGitApi();
```
You will have to add the collection to the url, e.g.:
https://redacted.visualstudio.com/DefaultCollection

**Comment by stevek-axo** [comment]
@stephenmichaelf Thanks! I got a workaround for fetching profile/team pictures by finding their IDs and firing an authenticated HTTP GET at https://redacted.visualstudio.com/_api/_common/identityImage?id=<redacted>

**Comment by stephenmichaelf** [comment]
@stevek-axo Glad you have something working! If you run into any other issues just open an issue in this repo and I can take a look :)

@alexkokkinos Were you able to make those changes and try again?

**Comment by alexkokkinos** [comment]
@stephenmichaelf Unfortunately, this stuff slid down in priority for me. I'll be able to try it sometime soon, probably in the next week or so. I'll definitely update here when I do.

**Comment by stephenmichaelf** [comment]
I am going to close this as we are removing the Identities API from the library.

@alexkokkinos If you have issues please open a new Issue with the specific problem, thanks!

**Comment by alexkokkinos** [comment]
Thanks @stephenmichaelf, I never really got around to trying it out again, but if it's being removed from the API it definitely makes sense to close this out.

**Comment by alexkokkinos** [comment]
@stephenmichaelf If I could ask one final question about this, since the Identities API will be removed, do you know where I might look for similar functionality so that I could report on identity-related information? Is there an alternative available right now?

**Comment by brunodefrancavalli** [comment]
> @stephenmichaelf Thanks! I got a workaround for fetching profile/team pictures by finding their IDs and firing an authenticated HTTP GET at https://redacted.visualstudio.com/_api/_common/identityImage?id=

I have seen this done in other places in the web, what is up with the ```_api``` in the URL instead of the documented ```_apis```? Is ```api``` legacy? Where do I find documentation about the calls on ```_api```?
